src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkBatchInsertRecord.java

```diff
@@ -0,0 +1,598 @@
+    }
+
+    /**
+     * Adds metadata for the given column in the file.
+     * 
+     * @param positionInTable
+     *            Indicates which column the metadata is for. Columns start at 1.
+     * @param name
+     *            Name for the column (optional if only using column ordinal in a mapping for SQLServerBulkCopy operation)
+     * @param jdbcType
+     *            JDBC data type of the column
+     * @param precision
+     *            Precision for the column (ignored for the appropriate data types)
+     * @param scale
+     *            Scale for the column (ignored for the appropriate data types)
+     * @param dateTimeFormatter
+     *            format to parse data that is sent
+     * @throws SQLServerException
+     *             when an error occurs
+     */
+    void addColumnMetadataInternal(int positionInTable,
+            String name,
+            int jdbcType,
+            int precision,
+            int scale,
+            DateTimeFormatter dateTimeFormatter) throws SQLServerException {
+        loggerExternal.entering(loggerClassName, "addColumnMetadata", new Object[] {positionInTable, name, jdbcType, precision, scale});
+
+        String colName = "";
+
+        if (0 >= positionInTable) {
+            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_invalidColumnOrdinal"));
+            Object[] msgArgs = {positionInTable};
+            throw new SQLServerException(form.format(msgArgs), SQLState.COL_NOT_FOUND, DriverError.NOT_SET, null);
+        }
+
+        if (null != name)
+            colName = name.trim();
+        else if ((columnNames != null) && (columnNames.length >= positionInTable))
+            colName = columnNames[positionInTable - 1];
+
+        if ((columnNames != null) && (positionInTable > columnNames.length)) {
+            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_invalidColumn"));
+            Object[] msgArgs = {positionInTable};
+            throw new SQLServerException(form.format(msgArgs), SQLState.COL_NOT_FOUND, DriverError.NOT_SET, null);
+        }
+
+        checkDuplicateColumnName(positionInTable, name);
+        switch (jdbcType) {
+            /*
+             * SQL Server supports numerous string literal formats for temporal types, hence sending them as varchar with approximate
+             * precision(length) needed to send supported string literals. string literal formats supported by temporal types are available in MSDN
+             * page on data types.
+             */
+            case java.sql.Types.DATE:
+            case java.sql.Types.TIME:
+            case java.sql.Types.TIMESTAMP:
+            case microsoft.sql.Types.DATETIMEOFFSET:
+                // The precision is just a number long enough to hold all types of temporal data, doesn't need to be exact precision.
+                columnMetadata.put(positionInTable, new ColumnMetadata(colName, jdbcType, 50, scale, dateTimeFormatter));
```

Comment:
Can we make the "50", which I'm assuming is the precision, a const so changing it is easier when necessary.

Reply:
The comment on line 217 explains why this is unnecessary.

Reply:
What if we create a temporal datatype in the future that accommodates precision > 50? If it's an arbitrarily assigned number than it can be just as arbitrarily changed.

Reply:
Precision here should be the one passed by user. I'm assuming 50 is used as precision based on the BulkCopy CSV implementation, it makes sense in case of CSV, because temporal types can be stored in CSV using any of the supported string literal format (https://docs.microsoft.com/en-us/sql/t-sql/data-types/datetime-transact-sql?view=sql-server-2017#supported-string-literal-formats-for-datetime) and we pass it as varchar/nvarchar to Server.